src/Swift.Bindings/src/Parser/BindingsGenerator.SwiftABIParser.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+using Swift.Runtime;
+
+namespace BindingsGeneration
+{
+    /// <summary>
+    /// Represents the root node of the ABI.
+    /// </summary>
+    public sealed class ABIRootNode
+    {
+        [JsonProperty("ABIRoot")]
+        public required RootNode ABIRoot { get; set; }
+    }
+
+    /// <summary>
+    /// Represents the root node of a module.
+    /// </summary>
+    public sealed class RootNode
+    {
+        public required string Kind { get; set; }
```

Comment:
Maybe enum would be a better long term type for `Kind` but I suppose we will move away from parsing Swift ABI so it is probably okay to use string for now.

Reply:
If `Kind` was implemented as enum, it would impact the JSON deserialization process, expecting `int` type instead of `string`:
```csharp
var abiRoot = JsonConvert.DeserializeObject<ABIRootNode>(jsonContent);
```

> will move away from parsing Swift ABI so it is probably okay to use string for now.

I think it is a good idea to wait until parser becomes more stable.